idseq_pipeline/commands/postprocess_functions.py

```diff
@@ -77,17 +79,32 @@ def accession2taxid(read_id, accession2taxid_dict, hit_type, lineage_map):
     species_taxid, genus_taxid, family_taxid = lineage_map.get(taxid, ("-100", "-200", "-300"))
     return species_taxid, genus_taxid, family_taxid
 
-def generate_taxid_fasta_from_accid(input_fasta_file, accession2taxid_path, lineagePath, output_fasta_file):
+def generate_taxid_fasta_from_accid(input_fasta_file, hit_summary_files, accession2taxid_path, lineagePath, output_fasta_file):
     accession2taxid_dict = shelve.open(accession2taxid_path)
     lineage_map = shelve.open(lineagePath)
+    nt_hit_summary_file = hit_summary_files[0]
+    nr_hit_summary_file = hit_summary_files[1]
+    def get_valid_hit(read_id, hit_summary_file):
+        hit_summary = subprocess.check_output("grep '^%s\t' %s" % (read_id, hit_summary_file), shell="true").split("\n")
+        assert len(hit_summary) == 1, "More than 1 line for %s in %s" % (read_id, hit_summary_file)
+        hit_line_columns = hit_summary[0].split("\t")
+        hit_level = hit_line_columns[1]
```

Comment:
I'd say this is the place to convert to int, as it would provide an extra measure of validaton in context where the exception would be easier to debug.